packages/block-renderer/src/components/block-renderer-container.tsx

```diff
@@ -110,7 +110,7 @@ const ScaledBlockRendererContainer = ( {
 				transform: `scale(${ scale })`,
 				height: scaledHeight,
 				maxHeight:
-					maxHeight !== 'none' && ( contentHeight as number ) > maxHeight
+					maxHeight !== 'none' && ( contentHeight as number ) > ( maxHeight as number )
```

Comment:
I commented in another PR about this. Currently, the component only accepts "none" and maxHeight, but types maxHeight as string|number.

But beyond that, `maxHeight` itself can accept several things -- like px values, and this logic wouldn't work with that.